signature/rsa_buffer/verify.c

```diff
@@ -0,0 +1,116 @@
+/* rsa_pub_2048.h
+ *
+ * Copyright (C) 2006-2018 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+/* This file is an example of verifying an RSA signature.
+ * The signature is PKCS#1.5 formatted.
+ * Key and data are held in buffers.
+ * "signature.h", used by this program, can be generated using "sign.c".
+ */
+
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/rsa.h>
+#include <wolfssl/wolfcrypt/sha256.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/asn_public.h>
+
+#include "rsa_pub_2048.h"
+#include "signature.h"
+
+/* Maximum bound on digest algorithm encoding around digest */
+#define MAX_ENC_ALG_SZ      32
+
+/* Main entry point.
+ * Verifies the signature with the message and RSA public key.
+ *
+ * argc  [in]  Count of command line arguments.
+ * argv  [in]  Command line argument vector.
+ * Returns 0 on success and 1 otherwise.
+ */
+int main(int argc, char* argv[])
+{
+    int            ret = 0;
+    Sha256         sha256;
+    Sha256*        pSha256;
```

Comment:
Can we initialize to NULL the pSha256, and pRsaKey just like you did in sign.c? (certain compilers reporting on this when checked at line 109 and 111, if condition on garbage value)

Reply:
fixed